cup-maven-plugin/src/test/java/de/jflex/plugin/cup/GenerateMojoTest.java

```diff
@@ -0,0 +1,102 @@
+package de.jflex.plugin.cup;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.ArgumentMatchers.endsWith;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+
+import java.io.File;
+import org.apache.maven.model.Model;
+import org.apache.maven.plugin.logging.Log;
+import org.apache.maven.plugin.testing.stubs.MavenProjectStub;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+/** Tests for {@link GenerateMojo}. */
+public class GenerateMojoTest {
+
+  private static final String TEST_TARGET_GENERATED_CUP_DIRECTORY =
+      "/tmp/target/generated-sources/cup";
```

Comment:
Does this work on Windows platforms? (Not really that important if it doesn't since is for development only, but if there's a platform independent API for tmp files, it would be slightly preferable)

Reply:
Done.
Good point. I have no idea.Guava has a handy `Files.createTempDir()`.